### ISSUE TITLE
Not all screens are disabled

### ISSUE THREAD

**Issue by stdedos** [slgobinath/SafeEyes#207]
I am seeing a bug related to https://github.com/slgobinath/SafeEyes/issues/121

When I am on secondary monitor _according to BIOS priorities_ (because otherwise ~/.config/monitors.xml has no primary monitor set), Safe Eyes only appears on said monitor.

Blocking or otherwise work perfectly, it's just that one of the monitors (the 'Primary') appears active

```bash
sntentos@sntentos-OptiPlex-9010 [2017-10-23 15:22:20]:~$ neofetch
            .-/+oossssoo+/-.               sntentos@sntentos-OptiPlex-9010
        `:+ssssssssssssssssss+:`           -------------------------------
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 16.04.3 LTS x86_64
    .ossssssssssssssssssdMMMNysssso.       Host: OptiPlex 9010 01
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 4.10.0-37-generic
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 7 days, 32 mins
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 2263
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: bash 4.3.48
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 1680x1050, 1680x1050
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   DE: Unity
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: Compiz
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   WM Theme: Ambiance
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Theme: Ambiance [GTK2/3]
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Icons: Ubuntu-mono-dark [GTK2/3]
  +sssssssssdmydMMMMMMMMddddyssssssss+     Terminal: gnome-terminal
   /ssssssssssshdmNNNNmyNMMMMhssssss/      CPU: Intel i7-3770 (8) @ 3.900GHz
    .ossssssssssssssssssdMMMNysssso.       GPU: NVIDIA GeForce GT 640 OEM
      -+sssssssssssssssssyyyssss+-         Memory: 9165MiB / 32123MiB
        `:+ssssssssssssssssss+:` 
            .-/+oossssoo+/-.                                       

```

**Comment by slgobinath** [comment]
Thanks for reporting this issue. I couldn't reproduce this on my Linux Mint system. I will try further and let you know the fix.

Thanks

**Comment by tonymihay** [comment]
I have the same issue.

**Comment by stdedos** [comment]
Care to ... comment if you looked through same files, what kind of OS do you have, etc, etc?

**Comment by tonymihay** [comment]
```
            .-/+oossssoo+/-.               
        `:+ssssssssssssssssss+:`           ------------------
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 16.04.3 LTS x86_64
    .ossssssssssssssssssdMMMNysssso.       Host: H110M-S2H
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 4.4.0-98-generic
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 4 hours, 5 mins
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 2011
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: zsh 5.1.1
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 1920x1080, 1920x1080
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   DE: Unity
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: Compiz
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   WM Theme: Ambiance
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Theme: Ambiance [GTK2/3]
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Icons: Ubuntu-mono-dark [GTK2/3]
  +sssssssssdmydMMMMMMMMddddyssssssss+     Terminal: gnome-terminal
   /ssssssssssshdmNNNNmyNMMMMhssssss/      CPU: Intel i5-6600 (4) @ 3.900GHz
    .ossssssssssssssssssdMMMNysssso.       GPU: Intel Integrated Graphics
      -+sssssssssssssssssyyyssss+-         Memory: 12937MiB / 15960MiB
        `:+ssssssssssssssssss+:` 
            .-/+oossssoo+/-.                                       
```

**Comment by Asalle** [comment]
Two monitors, same issue: only the left one is blocked during the break. Weird, because everything was fine in the previous version.

```
            .-/+oossssoo+/-.               tux@tux-HP-EliteBook-840-G3 
        `:+ssssssssssssssssss+:`           -------------------------------- 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 17.04 x86_64 
    .ossssssssssssssssssdMMMNysssso.       Model: HP EliteBook 840 G3 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 4.10.0-38-generic 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 3 days, 4 hours, 12 mins 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 2416 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: bash 4.4.7 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 1920x1080, 1920x1080 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   DE: Unity:Unity7 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: Compiz 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   WM Theme: Ambiance 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Theme: Ambiance [GTK2/3] 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Icons: Papirus [GTK2/3] 
  +sssssssssdmydMMMMMMMMddddyssssssss+     Terminal: gnome-terminal 
   /ssssssssssshdmNNNNmyNMMMMhssssss/      CPU: Intel i5-6300U (4) @ 3.0GHz 
    .ossssssssssssssssssdMMMNysssso.       GPU: Intel HD Graphics 520 
      -+sssssssssssssssssyyyssss+-         Memory: 4104MiB / 7862MiB 
        `:+ssssssssssssssssss+:` 
            .-/+oossssoo+/-.                                       

```

**Comment by slgobinath** [comment]
Hi all,
I couldn't reproduce this problem and unable to figure out the exact issue by manually analyzing the code. Could anyone try the following fix and let me know whether it resolves the issue?

1. Clone the master branch to your computer
```
git clone https://github.com/slgobinath/SafeEyes.git
```
2. Change the directory to SafeEyes
```
cd SafeEyes
```
3. Change the branch to fix
```
git checkout fix-multi-display-issue
```
4. Exit the currently running Safe Eyes instance.
5. Run Safe Eyes from source using the following command and see whether it fixes the problem. (Please try both manual break (Take a break now option) and the automatic break after 15 minutes interval)
```
python3 -m safeeyes --debug
```

Thanks in advance.

**Comment by stdedos** [comment]
Well, for me it would have to be Monday ... so enjoy your weekend :-D

**Comment by tonymihay** [comment]
Hi @slgobinath,
The issue is gone now.

**Comment by stdedos** [comment]
@slgobinath I seem to agree too :/

I will keep this version and "try" to replicate it though in the meantime

**Comment by slgobinath** [comment]
Hi all,
Thanks for the update.
I have made one more modification to this fix since it happens only in Unity desktop environment. I leave this branch unmerged until the next release for your testing purposes.

To pull the recent change, use `git pull origin fix-multi-display-issue` command after `git checkout fix-multi-display-issue` command.

**Comment by stdedos** [comment]
Small, somewhat rookie question:

Which channel would you recommend us following?
apt-get / pip / ./setup.py ?

**Comment by slgobinath** [comment]
I recommend `apt-get` for Ubuntu and its distributions because it can resolve all the dependencies.

`pip` is for those who do not have official packages (i.e except Ubuntu, Arch, and Gentoo users). The problem with `pip` is: it cannot install non-Python dependencies. There can be compatibility issues too because pip usually installs the latest version of libraries which may not be supported by the OS especially in non-rolling release systems like Ubuntu.

`setup.py` is only for those willing to use development version. I do not recommend it unless you want to test something.

Have you came across the same problem later while running the `fix-multi-display-issue` branch? If not, I will merge it with master branch and release the next version including this fix.

Thanks

**Comment by stdedos** [comment]
@slgobinath so far no, I consider it to be safe enough.

Of course, I'd prefer if all followers gave a +1 and not "depend" on me for that :smile:

**Comment by stdedos** [comment]
@slgobinath _Not meaning to be pushy_: When is the merge commit expected to reach apt-get repo? Or is it auto-synced once in origin/master?

**Comment by slgobinath** [comment]
Hi,
It is already merged into the master branch but I did not release it since I was on holidays in last few days and if there were any issues with releases I couldn't fix them immediately.

Today I am going to release the new version.

**Comment by slgobinath** [comment]
Fix released in 2.0.2 (2.0.1 has a dependency issue) but Launchpad is waiting for a day to build :-(
Once the package is built, I will release it to all Ubuntu versions.

**Comment by slgobinath** [comment]
Hi @stdedos,
The Ubuntu build takes days to build (Still in the waiting queue). I am disappointed and installed using pip on my computer.

**Comment by stdedos** [comment]
Why do the server has to build it, and not you provide the "packages"? :/

That's a big bottleneck right there ...

**Comment by slgobinath** [comment]
That's how Ubuntu Launchpad works.
Only the changes made to the code are uploaded to Launchpad and they build from the source in their servers. Usually, it takes only 1-2 hours but this time it is more than 3 days but still waiting...

**Comment by slgobinath** [comment]
Hi @stdedos,
Now, 2.0.2 is available in Ubuntu PPA.

**Comment by tonymihay** [comment]
Weird, but still not working for me. I'm on 2.0.2. 
![img_3973](https://user-images.githubusercontent.com/5081912/35040690-cdd02658-fb8a-11e7-94a4-0c7fec820227.jpg)

**Comment by stdedos** [comment]
@slgobinath Weird ... neither does it work for me anymore :/

**Comment by slgobinath** [comment]
Okay.
Though I am unable to reproduce it in my system, I will try my best to analyze the code and fix the issue.
@stdedos and @tonymihay  are you using Unity or Gnome?

**Comment by stdedos** [comment]
I doubt how much helpful I can be, since you already got my `neofetch` from above (i.e. Unity)

```
sntentos@optiplex-9010 [2018-01-22 11:20:28]:$ neofetch 
            .-/+oossssoo+/-.               sntentos@optiplex-9010
        `:+ssssssssssssssssss+:`           ----------------------
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 16.04.3 LTS x86_64
    .ossssssssssssssssssdMMMNysssso.       Host: OptiPlex 9010 01
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 4.13.0-26-generic
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 3 days, 23 hours
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 2365
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: bash 4.3.48
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 1680x1050, 1920x1080, 1920x1080
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   DE: Unity
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: Compiz
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   WM Theme: Ambiance
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Theme: Ambiance [GTK2/3]
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Icons: Ubuntu-mono-dark [GTK2/3]
  +sssssssssdmydMMMMMMMMddddyssssssss+     Terminal: gnome-terminal
   /ssssssssssshdmNNNNmyNMMMMhssssss/      CPU: Intel i7-3770 (8) @ 3.900GHz
    .ossssssssssssssssssdMMMNysssso.       GPU: NVIDIA GeForce GT 640 OEM
      -+sssssssssssssssssyyyssss+-         Memory: 9503MiB / 32119MiB
        `:+ssssssssssssssssss+:` 
            .-/+oossssoo+/-.                                   
```

**Comment by slgobinath** [comment]
@stdedos I made my fix for Unity. That's why I asked again for confirmation.
I will try my best.
Thanks

**Comment by tonymihay** [comment]
@slgobinath I use Unity.

**Comment by stdedos** [comment]
@slgobinath Unless I take a break from the leftmost screen (1680x1050), there is a one-off issue:
![image](https://user-images.githubusercontent.com/133706/35279721-888d8466-0056-11e8-97b8-d76ab457f4bf.png)

**Comment by tonymihay** [comment]
Hi @slgobinath,

Unfortunately the latest fix is not working even for two monitors.

**Comment by slgobinath** [comment]
Hi,
I am unable to produce this bug in my Linux Mint 18.3 with Cinnamon laptop.
In Safe Eyes 2.0.5, I have included additional logs to track moving break screen to specific displays. Could you please start Safe Eyes 2.0.5 using the following command and share the log after a break.
```
safeeyes --debug
```

The log should contain something like this:
```
2018-03-19 18:12:46,295 [INFO]:[MainThread] Show break screens in 1 display(s)
2018-03-19 18:12:46,338 [INFO]:[MainThread] Moved break screen to Display[0, 0]
```

**Comment by stdedos** [comment]
@slgobinath IMHO 2.0.5 fixes it correctly. I activated each screen multiple times, and breaks seem to come correctly on the correct screens (see attached log). 

[safeeyes.log](https://github.com/slgobinath/SafeEyes/files/1828651/safeeyes.log)

Is it possible that you log all of these entries regardless?
Log guru's can help you with that, and decide what's best (since I'd do the most stupid thing anyway).
_However, if no viable solution exists_, at least write them in `/tmp/safeeyes-*.log` or something - so that we don't have to go out of our way to help you, and we can report happenings on the first try. 10kbs for ~6 breaks don't seem like a waste of Disk/RAM space :wink: 

It would've been helpful to log where does your program "gets drawn" in the first place - if you had those entries, then triangulating this would be faster for everyone.

And, after all the "schooling" ... thank you for such a helpful program :smile:
My eyes thank you

**Comment by radek-sprta** [comment]
Yea, it might be a good idea to turn on debug mode by default, especially since SafeEyes already uses RotatingFileHandler.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by slgobinath** [comment]
Great!!!
Currently, logs are written to home directory in debug mode. I will enable debug mode from next version.

@tonymihay could you please confirm the fix.

Please close this issue if it has been fixed.

**Comment by tonymihay** [comment]
Hi @slgobinath,
The issues is gone. Thanx!

**Comment by stdedos** [comment]
@slgobinath I assume you mean `${HOME}/.config/safeeyes`? :smiley_cat:

**Comment by slgobinath** [comment]
It was in older versions. As a solution to this [Safe Eyes Debian bug report](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=881174), writing logs was limited to debug mode and the location was changed to `/home/<user>/safeeyes.log`.

 Currently, if you execute Safe Eyes in debug mode, you will get `~/safeeyes.log` file.

**Comment by stdedos** [comment]
I was talking about creating the log file regardless and prune it every week ... but I guess that's a bigger task.
_You should consider this making it in your backlog though_

**Comment by slgobinath** [comment]
Pruning is already taken care by the Python logger library I use.
Will create a log file in `tmp` directory from next version onwards.

**Comment by Asalle** [comment]
weirdly this issue affects me again on different laptop and os, now on GNOME. Apart from that, I cannot see the tray icon.
```
➜  ~ neofetch         
          /:-------------:\           asalle@myawesomelaptop 
       :-------------------::         ---------------------- 
     :-----------/shhOHbmp---:\       OS: Fedora 29 (Workstation Edition) x86_64 
   /-----------omMMMNNNMMD  ---:      Host: XPS 13 9370 
  :-----------sMMMMNMNMP.    ---:     Kernel: 4.18.16-300.fc29.x86_64 
 :-----------:MMMdP-------    ---\    Uptime: 17 mins 
,------------:MMMd--------    ---:    Packages: 1893 (rpm) 
:------------:MMMd-------    .---:    Shell: zsh 5.6.2 
:----    oNMMMMMMMMMNho     .----:    Resolution: 1920x1080, 2560x1440 
:--     .+shhhMMMmhhy++   .------/    DE: GNOME 3.30.1 
:-    -------:MMMd--------------:     Theme: Adwaita [GTK2/3] 
:-   --------/MMMd-------------;      Icons: Adwaita [GTK2/3] 
:-    ------/hMMMy------------:       Terminal: gnome-terminal 
:-- :dMNdhhdNMMNo------------;        CPU: Intel i7-8550U (8) @ 4.000GHz 
:---:sdNMMMMNds:------------:         GPU: Intel UHD Graphics 620 
:------:://:-------------::           Memory: 2182MiB / 15758MiB 
:---------------------://
                                                              

➜  ~ safeeyes --version
safeeyes 2.0.8
```

**Comment by slgobinath** [comment]
Hi @Asalle,

> I cannot see the tray icon

Do you use the [kStatusNotifier](https://extensions.gnome.org/extension/615/appindicator-support/) Gnome Shell plugin?

**Comment by Asalle** [comment]
> Do you use the [kStatusNotifier](https://extensions.gnome.org/extension/615/appindicator-support/) Gnome Shell plugin?

No I don't and when I tried to install it, it totally smashed my fedora installation, I think.

**Comment by slgobinath** [comment]
TopIconsPlus also works to an extent but not recommended as it has some issues. Anyway, no tray icon in Gnome is a design decision made by Gnome so we have to go with it. Safe Eyes supports RPC API which can be used in future to develop a Gnome plugin. For the moment, you can control Safe Eyes using command line arguments. Execute `safeeyes --help` for more details.

**Comment by Asalle** [comment]
ok, no icon is not a big issue
now, when I reinstalled my smashed fedora, I can't get safeeyes to show me the settings dialog. It freezed and after 2 mins I Ctrl-C'ed it.

```
[asalle@localhost ~]$ safeeyes --settings
^CTraceback (most recent call last):
  File "/home/asalle/.local/bin/safeeyes", line 11, in <module>
    sys.exit(main())
  File "/home/asalle/.local/lib/python3.7/site-packages/safeeyes/__main__.py", line 124, in main
    rpc_client.show_settings()
  File "/home/asalle/.local/lib/python3.7/site-packages/safeeyes/rpc.py", line 77, in show_settings
    self.proxy.show_settings()
  File "/usr/lib64/python3.7/xmlrpc/client.py", line 1112, in __call__
    return self.__send(self.__name, args)
  File "/usr/lib64/python3.7/xmlrpc/client.py", line 1452, in __request
    verbose=self.__verbose
  File "/usr/lib64/python3.7/xmlrpc/client.py", line 1154, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/usr/lib64/python3.7/xmlrpc/client.py", line 1166, in single_request
    http_conn = self.send_request(host, handler, request_body, verbose)
  File "/usr/lib64/python3.7/xmlrpc/client.py", line 1279, in send_request
    self.send_content(connection, request_body)
  File "/usr/lib64/python3.7/xmlrpc/client.py", line 1309, in send_content
    connection.endheaders(request_body)
  File "/usr/lib64/python3.7/http/client.py", line 1224, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.7/http/client.py", line 1016, in _send_output
    self.send(msg)
  File "/usr/lib64/python3.7/http/client.py", line 956, in send
    self.connect()
  File "/usr/lib64/python3.7/http/client.py", line 928, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/usr/lib64/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
KeyboardInterrupt

```

**Comment by slgobinath** [comment]
I experienced a similar issue on Fedora with TopIconsPlus. Do you have TopIconsPlus installed?

**Comment by Asalle** [comment]
No I don't have TopIconsPlus
```
[asalle@localhost ~]$ ls /usr/share/gnome-shell/extensions
alternate-tab@gnome-shell-extensions.gcampax.github.com
apps-menu@gnome-shell-extensions.gcampax.github.com
background-logo@fedorahosted.org
launch-new-instance@gnome-shell-extensions.gcampax.github.com
places-menu@gnome-shell-extensions.gcampax.github.com
window-list@gnome-shell-extensions.gcampax.github.com
[asalle@localhost ~]$ ls /usr/local/share/gnome-shell/extensions
ls: cannot access '/usr/local/share/gnome-shell/extensions': No such file or directory
[asalle@localhost ~]$  ls  ~/.local/share/gnome-shell/extensions
[asalle@localhost ~]$ 
```

**Comment by Asalle** [comment]
ok, this may be interesting for you @slgobinath :
I installed in /usr/local, then I installed in ~/.local/ and it didn't work at all, just froze
Then I uninstalled it from both places and installed it again in ~/.local, rebooted and it's all fine.
Only one screen is still blocked though.

**Comment by AdrienLemaire** [comment]
Hi there.
I'm facing the same issue (laptop + 2 external monitors). safeeyes will only block the focused screen, leaving the other 2 screens active and unblocked.
I've realized that I unconsciously move my mouse to a screen I don't care about when I get the notification, and I keep working during the break, which defeats the purpose of the app.

```console
$ neofetch
                   -`
                  .o+`
                 `ooo/                   dori@xps7390
                `+oooo:                  ------------
               `+oooooo:                 OS: Arch Linux x86_64
               -+oooooo+:                Host: XPS 13 7390 2-in-1
             `/:-:++oooo+:               Kernel: 5.5.2-arch1-1
            `/++++/+++++++:              Uptime: 4 days, 19 hours, 1
           `/++++++++++++++:             Packages: 1053 (pacman), 4
          `/+++ooooooooooooo/`           Shell: zsh 5.7.1
         ./ooosssso++osssssso+`          Resolution: 1920x1200, 1920
        .oossssso-````/ossssss+`         WM: sway
       -osssssso.      :ssssssso.        Theme: Tomorrow Night [GTK3
      :osssssss/        osssso+++.       CPU: Intel i7-1065G7 (8) @
     /ossssssss/        +ssssooo/-       GPU: Intel Iris Plus Graphics G7
   `/ossssso+/:-        -:/+osssso+-     Memory: 3279MiB / 31889MiB
  `+sso+:-`                 `.-/+oso:
 `++:.                           `-/+/
 .`                                 `/
```
